src/arrayMethodJoin.js

```diff
@@ -4,8 +4,35 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    if (this.length === 0) {
+      return resultString;
+    }
+
+    if (this.length === 1) {
+      resultString += this[0];
+
+      return resultString;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (separator === '') {
```

Comment:
Checking if separator is empty inside the loop is inefficient. This check should be done before the loop to avoid unnecessary comparisons.
So just modify the whole loop body to the code below

You need only to checks
```
      if (this[i] !== null && this[i] !== undefined) {
        resultString += this[i];
      }

      if (i < this.length - 1) {
        resultString += separator;
      }
```